pkg/chainaccessor/legacy_accessor.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/query/primitives"
+
+	"github.com/smartcontractkit/chainlink-ccip/pkg/consts"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/contractreader"
+	"github.com/smartcontractkit/chainlink-ccip/pkg/logutil"
+	cciptypes "github.com/smartcontractkit/chainlink-ccip/pkg/types/ccipocr3"
+)
+
+// LegacyAccessor is an implementation of cciptypes.ChainAccessor that allows the CCIPReader
+// to cutover and migrate away from depending directly on contract reader and contract writer.
+type LegacyAccessor struct {
+	lggr           logger.Logger
+	chainSelector  cciptypes.ChainSelector
+	contractReader contractreader.Extended
+	contractWriter types.ContractWriter
+	addrCodec      cciptypes.AddressCodec
+}
+
+var _ cciptypes.ChainAccessor = (*LegacyAccessor)(nil)
+
+func NewLegacyAccessor(
+	lggr logger.Logger,
+	chainSelector cciptypes.ChainSelector,
+	contractReader contractreader.Extended,
+	contractWriter types.ContractWriter,
+	addrCodec cciptypes.AddressCodec,
+) cciptypes.ChainAccessor {
+	return &LegacyAccessor{
+		lggr:           lggr,
+		chainSelector:  chainSelector,
+		contractReader: contractReader,
+		contractWriter: contractWriter,
+		addrCodec:      addrCodec,
+	}
+}
+
+func (l *LegacyAccessor) Metadata() cciptypes.AccessorMetadata {
+	allBindings := l.contractReader.GetAllBindings()
+	contracts := make(map[string]cciptypes.UnknownAddress, len(allBindings))
+	for contractName, binding := range allBindings {
+		addressBytes, err := l.addrCodec.AddressStringToBytes(binding[0].Binding.Address, l.chainSelector)
+		if err != nil {
+			l.lggr.Errorf("failed to convert address to bytes : %v", err)
+			continue
+		}
+		contracts[contractName] = addressBytes
+	}
+
+	return cciptypes.AccessorMetadata{
+		ChainSelector: l.chainSelector,
+		Contracts:     contracts,
+	}
+}
+
+func (l *LegacyAccessor) GetContractAddress(contractName string) ([]byte, error) {
+	bindings := l.contractReader.GetBindings(contractName)
+	if len(bindings) != 1 {
+		return nil, fmt.Errorf("expected one binding for the %s contract, got %d", contractName, len(bindings))
+	}
+
+	addressBytes, err := l.addrCodec.AddressStringToBytes(bindings[0].Binding.Address, l.chainSelector)
+	if err != nil {
+		return nil, fmt.Errorf("convert address %s to bytes: %w", bindings[0].Binding.Address, err)
+	}
+
+	return addressBytes, nil
+}
+
+func (l *LegacyAccessor) GetChainFeeComponents(ctx context.Context) (cciptypes.ChainFeeComponents, error) {
+	fc, err := l.contractWriter.GetFeeComponents(ctx)
+	if err != nil {
+		return cciptypes.ChainFeeComponents{}, fmt.Errorf("get fee components: %w", err)
+	}
+
+	return *fc, nil
+}
+
+func (l *LegacyAccessor) Sync(ctx context.Context, contracts cciptypes.ContractAddresses) error {
+	// TODO(NONEVM-1865): implement
+	panic("implement me")
+}
+
```

Comment:
You could define `SendRequestedEvent` here or make another file for the accessor types. In the future it would be great to get the type directly from the go bindings, but I don't think we should do that until the accessor is defined outside of the plugin.